plugins/redirects.ts

```diff
@@ -98,16 +99,17 @@ function parseRedirection(
 function html(redirects: Redirect[], site: Site): void {
   for (const [url, to, statusCode] of redirects) {
     const timeout = (statusCode === 301 || statusCode === 308) ? 0 : 1;
+    const revisedTo = joinUrl(site.options.location, to).pathname;
```

Comment:
The `joinUrl` dependency is not needed. You can get the final url with `site.url(to)`.

Maybe the path resolution should be moved to `parseRedirection` function, so it's available for all redirects methods, not only HTML.

Reply:
Nice tips and background information.